actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -139,6 +141,34 @@ def negotiate_mime(order)
         order.include?(Mime::ALL) ? format : nil
       end
 
+      class VariantInquirer # :nodoc:
+        delegate :==, :each, :empty?, to: :variants
+
+        def initialize(variants = [])
+          @variants = variants
+        end
+
+        def any?(*candidates)
+          (variants & candidates).any?
+        end
+
+        def to_ary
+          variants
+        end
+
+        protected
+          attr_reader :variants
```

Comment:
It shouldn't be public because `request.variant.variants` is an awkward and opaque invocation for accessing the variants as an array.

I'd be open to exposing it publicly as `to_a`.